docs/docs/dev-plugins.mdx

```diff
@@ -0,0 +1,120 @@
+---
+id: dev-plugins
+title: Plugin development
+---
+
+Vitessce supports plugin view types, coordination types, and file types.
+
+Plugin view types and coordination types can enable development of interactive visualizations to support custom use cases (i.e., use cases beyond those supported by existing view types).
+Plugin file types can enable loading data that is stored in custom file formats.
+
+:::caution
+
+Note that we cannot guarantee that future versions of Vitessce will not break existing plugins.
+For instance, we do not make any guarantees that exported helper functions such as React data-loading hooks (e.g., `useRasterData`, `useExpressionMatrixData`, etc.) will not be removed or will have a consistent function signature across Vitessce versions.
+Therefore, __plugin developers should be sure to document which version(s) of Vitessce that plugins have been developed under__.
+
+:::
+
+:::tip
+
+If you believe a plugin would be applicable to the wider Vitessce user base, we encourage contributing it back to the Vitessce core via a pull request, which can also facilitate us to support the view type or file type in future Vitessce versions.
```

Comment:
Or just opening a discussion (rejecting PR's feels kinda mean sometimes, but maybe it's just my insecurity)

Reply:
don't want to tell people to code stuff just to reject to it

Reply:
Yes good point. Changed to "open a discussion" in https://github.com/vitessce/vitessce/pull/1212/commits/8964b41c74fd7782e5c7f7ba901ab8a2966b846b#diff-147263ad13f278914f887d2d07c0c9a9d4ac97a32fd4d8bf2363c5ba59bc6470L21